File: net/YapayGatewayLib/model/Transaction.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace YapayGatewayLib.Model
{
    public class Transaction
    {
        [JsonProperty(PropertyName = "codigoEstabelecimento")]
        string storeCode { get; set; }

        [JsonProperty(PropertyName = "codigoFormaPagamento")]
        int paymentCode { get; set; }

        [JsonProperty(PropertyName = "transacao")]
        TransactionData transactionData { get; set; }

        [JsonProperty(PropertyName = "checkout")]
        CheckoutData checkout { get; set; }

        [JsonProperty(PropertyName = "dadosCartao")]
        CreditCardData creditCard { get; set; }

        [JsonProperty(PropertyName = "dadosMultiplosCartoes")]
        List<CreditCardData> multipleCard { get; set; }

        [JsonProperty(PropertyName = "dadosDebito")]
        DebitCardData debitCard { get; set; }

        [JsonProperty(PropertyName = "itensDoPedido")]
        List<ItemData> items { get; set; }

        [JsonProperty(PropertyName = "dadosCobranca")]
        ChargingData charging { get; set; }

        [JsonProperty(PropertyName = "dadosEntrega")]
        DeliveryData delivery { get; set; }

        [JsonProperty(PropertyName = "dadosAirline")]
        AirlineData airline { get; set; }

        [JsonProperty(PropertyName = "camposExtras")]
        List<ExtraField> extraFields { get; set; }
    }
}

File: ruby/yapay_gw/lib/yapay_gw/models/transaction.rb
class Transaction
    attr_accessor :store_code
    attr_accessor :payment_code
    attr_accessor :transaction_data
    attr_accessor :transaction_checkout_data
    attr_accessor :transaction_card_data
    attr_accessor :transaction_multiple_card_data
    attr_accessor :transaction_item_data
    attr_accessor :transaction_charging_data
    attr_accessor :transaction_delivery_data
    attr_accessor :transaction_extra_fields

    def as_json(options = {})
        json = {:codigoEstabelecimento => store_code, 
                :codigoFormaPagamento => payment_code, 
                :transacao => transaction_data,
                :checkout => transaction_checkout_data,
                :dadosCartao => transaction_card_data,
                :dadosMultiplosCartoes => transaction_multiple_card_data,
                :itensDoPedido => transaction_item_data,
                :dadosCobranca => transaction_charging_data,
                :dadosEntrega => transaction_delivery_data,
                :camposExtras => transaction_extra_fields
            }
        json.delete_if { |k, v| v.nil? }
        json
    end
end